source/core/Array.cpp

```diff
@@ -987,30 +982,30 @@ VIREO_FUNCTION_SIGNATURE4(ArraySplit, TypedArrayCoreRef, TypedArrayCoreRef, Type
     return _NextInstruction();
 }
 
-struct Sort1DArrayInstruction : public InstructionCore
+struct Sort1DArrayInstruction : InstructionCore
 {
     _ParamDef(TypedArrayCoreRef, OutArray);
     _ParamDef(TypedArrayCoreRef, InArray);
     _ParamImmediateDef(InstructionCore*, Next);
-    inline InstructionCore* Snippet()   { return this + 1; }
-    inline InstructionCore* Next()      { return this->_piNext; }
+    InstructionCore* Snippet()   { return this + 1; }
+    InstructionCore* Next()      { return this->_piNext; }
 };
+
 // Emit the sort instruction for specific type
 //------------------------------------------------------------
 InstructionCore* EmitSortInstruction(ClumpParseState* pInstructionBuilder)
 {
-    ConstCStr pSortOpName = "Sort1DArrayInternal";
+    const ConstCStr pSortOpName = "Sort1DArrayInternal";
     SubString sortOpToken(pSortOpName);
 
     pInstructionBuilder->ReresolveInstruction(&sortOpToken);
-    InstructionCore* pInstruction = nullptr;
-    TypedArrayCoreRef arrayArg = *(TypedArrayCoreRef*)pInstructionBuilder->_argPointers[0];
+    TypedArrayCoreRef arrayArg = *static_cast<TypedArrayCoreRef*>(pInstructionBuilder->_argPointers[0]);
     TypeRef elementType = arrayArg->ElementType();
     SubString LTName("IsLTSort");
     // Add param slot to hold the snippet
-    Int32 snippetArgId = pInstructionBuilder->AddSubSnippet();
-    Sort1DArrayInstruction* sortOp = (Sort1DArrayInstruction*) pInstructionBuilder->EmitInstruction();
-    pInstruction = sortOp;
+    const Int32 snippetArgId = pInstructionBuilder->AddSubSnippet();
+    auto sortOp = static_cast<Sort1DArrayInstruction*>(pInstructionBuilder->EmitInstruction());
+    InstructionCore* p_instruction = sortOp;
```

Comment:
Why??  pInstruction is fine, and consistent with pInstructionBuilder, which you did not change.